### PR TITLE
 #164798276 Set up testing 

### PR DESCRIPTION
**Pull request by anyatibrian**
#### What does this PR do?
Sets up testing for for authors heaven
#### Description of Task to be completed?
- Add tests for user registration login .
- Add tests  for user login endpoint.
- Add various files required for the tests.
- Add configurations for Travis
- Add travis configuration with slack
- Add test for model and serializers

#### How should this be manually tested?
After cloning the repo, cd into it and run the following commands to activate the virtual environment
- `$ virtualenv venv`
- `$ source venv/bin/activate`

Install all requirements
- `$ pip install -r requirements.txt`

Run the tests
- `$ python manage.py test`

#### What are the relevant pivotal tracker stories?
[#164798276](https://www.pivotaltracker.com/story/show/164798276)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (inno-asiimwe) [inline .vscode/settings.json]: Do not push your editor settings.

**Comment by inno-asiimwe** [comment]
Looks like you have work from someone else's commit, please rectify this and rebase with develop as well.

**Comment by inno-asiimwe** [comment]
Since you are integrating Travis, ensure that you add a build badge to the `README`, also ensure that Travis is integrated with slack so that the team can receive notifications via slack.

File: .vscode/settings.json (hunk 1 above)
Comment (anyatibrian) [reply .vscode/settings.json]: finished thanks

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+language: python
+
+python:
+  - "3.6"
+
+services:
+  - postgresql
+
+env:
```
Comment (inno-asiimwe) [inline .travis.yml]: You can have this in the environment on travis.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,4 @@
+[![Build Status](https://travis-ci.org/andela/ah-backend-prime.svg?branch=develop)](https://travis-ci.org/andela/ah-backend-prime)
```
Comment (e-ian) [inline README.md:1]: @anyatibrian why does the travis badge indicate a failing build.

Comment (anyatibrian) [reply README.md:1]: its because my branch has not yet been merged to develop

File: authors/apps/authentication/tests/test_base.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+from django.urls import reverse
+from ...authentication.models import User
+
+
+class BaseTest(APITestCase):
+    """
+    sets up a base class upon which all our test depends on
+    """
+    def setUp(self):
+        self.clien= APIClient()
```
Comment (e-ian) [inline authors/apps/authentication/tests/test_base.py]: @anyatibrian in you BaseTest class under the method `def setUp` I notice you creating an instance of the `APIClient()` as `self.clien=APIClient()` and there is a typo it should be `self.client=APIClient()`, as you continue using `self.client` when mocking your tests.

Comment (anyatibrian) [reply authors/apps/authentication/tests/test_base.py]: its done, i just realised it few minutes ago

**Pull request merged**